src/renderer/components/query-result-table-virtualized.jsx

```diff
@@ -0,0 +1,388 @@
+        <div className="ui small label" title="Copy as" style={{"float": "right","margin": "3px"}}>
+          <i className="copy icon"></i>
+          <a className="detail" style={styleCopied}>Copied</a>
+          <a className="detail"
+            style={styleButtons}
+            onClick={() => onCopyToClipboardClick(rows, 'CSV')}>CSV</a>
+          <a className="detail"
+            style={styleButtons}
+            onClick={() => onCopyToClipboardClick(rows, 'JSON')}>JSON</a>
+        </div>
+      );
+    }
+
+    return (
+      <div style={{"height":"32px","width":tableWidth}}>
+        <div style={{"marginLeft":"10px","float":"left"}}>Rows: {rowCount}</div>
+        {copyPanel}
+      </div>
+    );
+  }
+
+  getColumnWidth({index}) {
+    var field = this.props.fields[index];
+    if ( field && this.state.columnWidths && this.state.columnWidths[field.name] !== undefined ) {
+      return this.state.columnWidths[field.name];
+    } else if ( this.state.autoColumnWidths && this.state.autoColumnWidths[index] !== undefined ) {
+      return this.state.autoColumnWidths[index];
+    } else {
+      return 50;
+    }
+  }
+
+  render() {
+    const { rowCount, fields, rows } = this.props;
+    const { tableWidth, tableHeight } = this.state;
+    const averageTableCellWidth = tableWidth / fields.length;
+
+    // not completed loaded yet
+    if (!tableWidth) {
+      return null;
+    }
+
+    var j = 0, totalColumnWidths = 0, autoColumnWidths = fields.map(({ name, index }) => this.resolveCellWidth(name, fields, rows, averageTableCellWidth))
+
+    for ( j=0; j < autoColumnWidths.length; j=j+1 ) {
+        totalColumnWidths = totalColumnWidths + autoColumnWidths[j];
+        if ( (j+1) === autoColumnWidths.length && totalColumnWidths < tableWidth ) {
+            totalColumnWidths = totalColumnWidths - autoColumnWidths[j];
+            autoColumnWidths[j] = (tableWidth - totalColumnWidths);
+        }
+    }
+
+    this.state.autoColumnWidths = autoColumnWidths;
+
+    let previewModal = null;
+    if (this.state.showPreview) {
+      previewModal = (
+        <PreviewModal
+          value={this.state.valuePreview}
+          onCloseClick={::this.onClosePreviewClick}
+        />
+      );
+    }
+
+    var self = this;
+
+    return (
+      <div>
+        {previewModal}
+
+        <ScrollSync>
+          {({ clientHeight, clientWidth, onScroll, scrollHeight, scrollLeft, scrollTop, scrollWidth }) => {
+
+             return ( <div>
+                
+                {this.renderHeader()}
+
+                {(() => {
+
+                    if ( fields.length > 0 ) {
+
+                        return <div style={{position:"relative"}}>
+
+                          <div className="scrollPlaceholder" 
+                            style={{left:(tableWidth - scrollbarSize()),
+                              width:scrollbarSize()}}></div>
+
+                          <Grid
+                              ref={(ref) => self.headerGrid = ref}
```

Comment:
I guess there is no need using self there since looks like you are only using arrow functions. So the context is always the one from the root function.